terraform/aws/modules/splunk-server/resources.tf

```diff
@@ -16,13 +16,98 @@ data "aws_ami" "latest-ubuntu" {
 }
 
 
+resource "aws_iam_role" "splunk_role" {
+  name = "splunk_role_${var.config.key_name}"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "ec2.amazonaws.com"
+      },
+      "Effect": "Allow",
+      "Sid": ""
+    }
+  ]
+}
+EOF
+
+}
+
+resource "aws_iam_instance_profile" "splunk_profile" {
+  name = "splunk_profile_${var.config.key_name}"
+  role = aws_iam_role.splunk_role.name
+}
+
+
+data "aws_iam_policy_document" "splunk_logging" {
+
+  statement {
+    actions = [
+      "logs:DescribeLogGroups",
+      "logs:DescribeLogStreams",
+      "logs:DescribeDestinations",
+      "logs:DescribeDestinations",
+      "logs:TestMetricFilter",
+    ]
+
+    resources = [
+      "*",
+    ]
+  }
+
+  statement {
+    actions = [
+      "logs:Describe*",
+      "logs:Get*",
+      "logs:FilterLogEvents",
+    ]
+
+    resources = [
+      "*",
+    ]
+  }
+
+  statement {
+    actions = [
+      "sqs:GetQueueAttributes",
+      "sqs:ListQueues",
+      "sqs:ReceiveMessage",
+      "sqs:GetQueueUrl",
+      "sqs:DeleteMessage",
+      "s3:Get*",
+      "s3:List*",
+      "s3:Delete*",
+      "kms:Decrypt",
+    ]
+
+    resources = [
+      "*"
```

Comment:
Just a drive-by comment, please feel free to ignore, but I really don't think it is appropriate to give this role broad permissions on `*` like this (it is quite a pet peeve of mine when amazon includes examples like this in their docs), especially when your terraform is already aware of the s3 bucket name and the sqs queue name/account

Reply:
@danielpops : Thank you, you make a good point here, i am trying to figure out what are the exact perms required? if you have any. recommendations i would love to hear it!